File: ccif/src/java/other/org/himalay/ccif/TerminalAssociation.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:06 EST 2016")

public  class TerminalAssociation extends BinStruct implements PublicBinMsg {




    // members variables
    // identifierList
    public ByteArray identifierList ;

    public TerminalAssociation () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize identifierList
        identifierList= new ByteArray();
        identifierList.setSizeType("FIRST_UI16");
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read identifierList
        {
            retVal+=identifierList.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for identifierList**/
        }

        // write identifierList
        {
            retVal += identifierList.write(ostream);
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("TerminalAssociation\n");
        dc.increaseIndent();
        int retVal= 0;
        // write identifierList
        dc.indent();
        dc.getPs().print("identifierList: "+identifierList.getSize()+"(0x"+Integer.toHexString(identifierList.getSize())+")\n");
        this.identifierList.dump(dc);
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for identifierList
    //public ByteArray getIdentifierList()
    //{
    //    return identifierList ;
    //}


    // Setter for identifierList
    //public void setIdentifierList(ByteArray val)
    //{
    //    this.identifierList= val;
    //}


    public void setIdentifierList(byte[] val)
    {
        this.identifierList.setData(val);
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/TerminalAssociation.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "TerminalAssociation.h"

#ifdef org_himalay_ccif_TerminalAssociation__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
TerminalAssociation::TerminalAssociation() {
    references= 0;


    // identifierList
    org_himalay_msgs_runtime_ByteArray_NEW(identifierList);
    identifierList->setSizeType("FIRST_UI16");
}
// Destructor
TerminalAssociation::~TerminalAssociation() {
    // identifierList
    org_himalay_msgs_runtime_ByteArray_DELETE( identifierList);
}

int TerminalAssociation::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // identifierList
    {
        retVal+=identifierList->read(istream);
    }
    return retVal;
}

int TerminalAssociation::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read identifierList
    {
        retVal+=identifierList->read(istream);
    }

    return retVal;
}

int TerminalAssociation::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for identifierList**/
    }

    int retVal= 0;
    // write identifierList
    {
        retVal += identifierList->write(ostream);
    }

    return retVal;
}

int TerminalAssociation::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("TerminalAssociation\n");
    dc.increaseIndent();
    int retVal= 0;
    // write identifierList
    dc.indent();
    dc.getPs().print("identifierList");
    identifierList->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for identifierList
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE TerminalAssociation::getIdentifierList()
//{
//return this->identifierList;
//}

// Setter for identifierList
//void TerminalAssociation::setIdentifierList(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->identifierList= val;
//}

int TerminalAssociation::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_TerminalAssociation__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::TerminalAssociation* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::TerminalAssociation* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code